### PR TITLE
- Fixes zonemaster/zonemaster-engine#516

### PR DESCRIPTION
**Pull request by vlevigneron**
- Fixes zonemaster/zonemaster-engine#516
- Add new messages for Nameserver02.
- Need Swedish and Danish translation to be revised.

### CODE REVIEW

File: lib/Zonemaster/Engine/Test/Nameserver.pm [hunk 1]
```diff
@@ -189,7 +191,7 @@ sub translation {
         ANSWER_BAD_RCODE    => 'Nameserver {ns}/{address} answered AAAA query with an unexpected rcode ({rcode}).',
         AXFR_AVAILABLE      => 'Nameserver {ns}/{address} allow zone transfer using AXFR.',
         AXFR_FAILURE        => 'AXFR not available on nameserver {ns}/{address}.',
-	BROKEN_EDNS_SUPPORT => 'Incorrect response from {ns}/{address} asking for {dname} when EDNS is used.',
+        BREAKS_ON_EDNS      => 'Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No Response).',
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Nameserver.pm]: 'No response from {ns}/{address} when EDNS is used in query asking for {dname}.'

File: lib/Zonemaster/Engine/Test/Nameserver.pm [hunk 2]
```diff
@@ -208,13 +210,14 @@ sub translation {
           . 'nameserver {ns}/{address} returns same RCODE "{rcode}".',
         DIFFERENT_SOURCE_IP => 'Nameserver {ns}/{address} replies on a SOA query with a different source address '
           . '({source}).',
-        NO_EDNS_SUPPORT            => 'Nameserver {ns}/{address} does not support EDNS0 (replies with FORMERR).',
+        EDNS_RESPONSE_WITHOUT_EDNS => 'Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No OPT record).',
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Nameserver.pm]: 'Response without EDNS from {ns}/{address} on query with EDNS0 asking for {dname}.'

File: lib/Zonemaster/Engine/Test/Nameserver.pm [hunk 3]
```diff
@@ -208,13 +210,14 @@ sub translation {
...
+        EDNS_VERSION_ERROR         => 'Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (EDNS version should be 0).',
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Nameserver.pm]: 'Incorrect version of EDNS (expected 0) in response from {ns}/{address} on query with EDNS (version 0) asking for {dname}.',

File: share/da.po [hunk 4]
```diff
@@ -788,9 +788,9 @@ msgstr "Navneserveren {ns}/{address} tillader zoneoverførsler via AXFR."
 msgid  "AXFR not available on nameserver {ns}/{address}."
 msgstr "AXFR er ikke tilgængeligt på navneserveren {ns}/{address}."
 
-#: NAMESERVER:BROKEN_EDNS_SUPPORT
-msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used."
-msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used."
+#: NAMESERVER:BREAKS_ON_EDNS
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No Response)."
+msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No Response)."
```
Comment (matsduf) [inline share/da.po]: 'No response from {ns}/{address} when EDNS is used in query asking for {dname}.'

File: share/da.po [hunk 5]
```diff
@@ -865,6 +865,15 @@ msgstr ""
 "Navneserveren {ns}/{address} svarede på en SOA-forespørgsel fra en anden "
 "kilde ({source})."
 
+#: NAMESERVER:EDNS_RESPONSE_WITHOUT_EDNS
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No OPT record)."
+msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No OPT record)."
```
Comment (matsduf) [inline share/da.po]: 'Response without EDNS from {ns}/{address} on query with EDNS0 asking for {dname}.'

(Danish in a later PR, create issue on @matsduf)

File: share/da.po [hunk 6]
```diff
@@ -865,6 +865,15 @@ msgstr ""
...
+
+
+#: NAMESERVER:EDNS_VERSION_ERROR
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (EDNS version should be 0)."
+msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (EDNS version should be 0)."
```
Comment (matsduf) [inline share/da.po]: 'Incorrect version of EDNS (expected 0) in response from {ns}/{address} on query with EDNS (version 0) asking for {dname}.'

(Danish in a later PR, create issue on @matsduf)

File: share/en.po [hunk 7]
```diff
@@ -752,9 +752,9 @@ msgstr "Nameserver {ns}/{address} allow zone transfer using AXFR."
 msgid  "AXFR not available on nameserver {ns}/{address}."
 msgstr "AXFR not available on nameserver {ns}/{address}."
 
-#: NAMESERVER:BROKEN_EDNS_SUPPORT
-msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used."
-msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used."
+#: NAMESERVER:BREAKS_ON_EDNS
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No Response)."
+msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No Response)."
```
Comment (matsduf) [inline share/en.po]: 'No response from {ns}/{address} when EDNS is used in query asking for {dname}.'

File: share/en.po [hunk 8]
```diff
@@ -829,6 +829,14 @@ msgstr ""
 "Nameserver {ns}/{address} replies on a SOA query with a different source "
 "address ({source})."
 
+#: NAMESERVER:EDNS_RESPONSE_WITHOUT_EDNS
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No OPT record)."
+msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No OPT record)."
```
Comment (matsduf) [inline share/en.po]: 'Response without EDNS from {ns}/{address} on query with EDNS0 asking for {dname}.'

File: share/en.po [hunk 9]
```diff
@@ -829,6 +829,14 @@ msgstr ""
...
+
+#: NAMESERVER:EDNS_VERSION_ERROR
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (EDNS version should be 0)."
+msgstr "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (EDNS version should be 0)."
```
Comment (matsduf) [inline share/en.po]: 'Incorrect version of EDNS (expected 0) in response from {ns}/{address} on query with EDNS (version 0) asking for {dname}.'

File: share/sv.po [hunk 10]
```diff
@@ -802,7 +802,7 @@ msgid  "AXFR not available on nameserver {ns}/{address}."
 msgstr "AXFR är inte tillgänglig från namnservern {ns}/{address}."
 
 #: NAMESERVER:BROKEN_EDNS_SUPPORT
-msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used."
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No Response)."
```
Comment (matsduf) [inline share/sv.po]: 'No response from {ns}/{address} when EDNS is used in query asking for {dname}.'

'Inget svar från {ns}/{address} på förfrågan med EDNS (fråga efter {dname}).'

File: share/fr.po [hunk 11]
```diff
@@ -924,8 +924,8 @@ msgstr ""
 "noms {ns}/{address}."
 
 #: NAMESERVER:BROKEN_EDNS_SUPPORT
-msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used."
-msgstr "Réponse incorrecte de {ns}/{address} sur le nom de domaine '{dname}' lorsque EDNS est utilisé."
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No Response)."
+msgstr "Réponse incorrecte de {ns}/{address} sur le nom de domaine '{dname}' lorsque EDNS est utilisé (Aucune réponse)."
```
Comment (matsduf) [inline share/fr.po]: #: NAMESERVER:BREAKS_ON_EDNS

'No response from {ns}/{address} when EDNS is used in query asking for {dname}.'

Plus French.

File: share/fr.po [hunk 12]
```diff
@@ -1006,6 +1006,14 @@ msgstr ""
 "Le serveur de noms {ns}/{address} répond depuis une autre adresse que celle "
 "à laquelle la requête a été envoyée."
 
+#: NAMESERVER:EDNS_RESPONSE_WITHOUT_EDNS
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No OPT record)."
+msgstr "Réponse incorrecte de {ns}/{address} sur le nom de domaine '{dname}' lorsque EDNS est utilisé (Pas d'enregistrement de type OPT)."
```
Comment (matsduf) [inline share/fr.po]: 'Response without EDNS from {ns}/{address} on query with EDNS0 asking for {dname}.'

Plus French.

File: share/fr.po [hunk 13]
```diff
@@ -1006,6 +1006,14 @@ msgstr ""
...
+
+#: NAMESERVER:EDNS_VERSION_ERROR
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (EDNS version should be 0)."
+msgstr "Réponse incorrecte de {ns}/{address} sur le nom de domaine '{dname}' lorsque EDNS est utilisé (La version EDNS devrait être 0)."
```
Comment (matsduf) [inline share/fr.po]: 'Incorrect version of EDNS (expected 0) in response from {ns}/{address} on query with EDNS (version 0) asking for {dname}.'

Plus French.

File: share/sv.po [hunk 14]
```diff
@@ -878,6 +878,14 @@ msgstr ""
 "Namnservern {ns}/{address} besvarade en SOA-fråga från en annan källadress "
 "({source})."
 
+#: NAMESERVER:EDNS_RESPONSE_WITHOUT_EDNS
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (No OPT record)."
+msgstr "Felaktigt DNS-svar från {ns}/{address} när {dname} efterfrågas med EDNS."
```
Comment (matsduf) [inline share/sv.po]: 'Response without EDNS from {ns}/{address} on query with EDNS0 asking for {dname}.'

'Svar utan EDNS från {ns}/{address} på  förfrågan med EDNS0 (fråga efter {dname}).'

File: share/sv.po [hunk 15]
```diff
@@ -878,6 +878,14 @@ msgstr ""
...
+
+#: NAMESERVER:EDNS_VERSION_ERROR
+msgid  "Incorrect response from {ns}/{address} asking for {dname} when EDNS is used (EDNS version should be 0)."
+msgstr "Felaktigt DNS-svar från {ns}/{address} när {dname} efterfrågas med EDNS."
```
Comment (matsduf) [inline share/sv.po]: 'Incorrect version of EDNS (expected 0) in response from {ns}/{address} on query with EDNS (version 0) asking for {dname}.'

'Felaktig EDNS-version (förväntad version var 0) i svar från {ns}/{address} på förfrågan med EDNS version 0 (fråga efter {dname}).'

File: lib/Zonemaster/Engine/Test/Nameserver.pm (hunk 1 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Nameserver.pm]: Done

File: lib/Zonemaster/Engine/Test/Nameserver.pm (hunk 2 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Nameserver.pm]: Done

File: lib/Zonemaster/Engine/Test/Nameserver.pm (hunk 3 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/Nameserver.pm]: Done

File: share/fr.po (hunk 11 above)
Comment (vlevigneron) [reply share/fr.po]: Done

File: share/fr.po (hunk 13 above)
Comment (vlevigneron) [reply share/fr.po]: Done

File: share/sv.po (hunk 14 above)
Comment (vlevigneron) [reply share/sv.po]: Done

File: share/fr.po (hunk 12 above)
Comment (vlevigneron) [reply share/fr.po]: Done

File: share/sv.po (hunk 15 above)
Comment (vlevigneron) [reply share/sv.po]: Done

File: share/da.po (hunk 4 above)
Comment (vlevigneron) [reply share/da.po]: Done

File: share/da.po (hunk 5 above)
Comment (vlevigneron) [reply share/da.po]: Done

File: share/da.po (hunk 6 above)
Comment (vlevigneron) [reply share/da.po]: See zonemaster/zonemaster-engine#518

File: share/da.po (hunk 4 above)
Comment (matsduf) [reply share/da.po]: The msgid must be updated to: 
'No response from {ns}/{address} when EDNS is used in query asking for {dname}.'

Set msgstr to the same.

File: share/da.po (hunk 5 above)
Comment (matsduf) [reply share/da.po]: msgid must be set to:
'Response without EDNS from {ns}/{address} on query with EDNS0 asking for {dname}.'

Set msgstr to the same.

File: share/da.po (hunk 6 above)
Comment (matsduf) [reply share/da.po]: Msgid must be set to:

'Incorrect version of EDNS (expected 0) in response from {ns}/{address} on query with EDNS (version 0) asking for {dname}.'

Set msgstr to the same.

File: share/sv.po (hunk 10 above)
Comment (vlevigneron) [reply share/sv.po]: Done

**Comment by matsduf** [comment]
For some reason my three last comments show up as resolved when they are not.

**Comment by vlevigneron** [comment]
@matsduf the issues on da.po has been resolved since the corresponding issue has been created (zonemaster/zonemaster-engine#518)

File: share/da.po (hunk 4 above)
Comment (vlevigneron) [reply share/da.po]: Done

File: share/da.po (hunk 5 above)
Comment (vlevigneron) [reply share/da.po]: Done

File: share/da.po (hunk 6 above)
Comment (vlevigneron) [reply share/da.po]: Done

**Pull request merged**